src/cryptography/hazmat/primitives/asymmetric/dh.py

```diff
@@ -99,3 +92,71 @@ def __ne__(self, other):
 
     p = utils.read_only_property("_p")
     g = utils.read_only_property("_g")
+
+
+@six.add_metaclass(abc.ABCMeta)
+class DHParameters(object):
+    pass
+
+
+@six.add_metaclass(abc.ABCMeta)
+class DHParametersWithNumbers(DHParameters):
+    @abc.abstractmethod
+    def parameter_numbers(self):
+        """
+        Returns a DHParameterNumbers.
+        """
+
+
+@six.add_metaclass(abc.ABCMeta)
+class DHPrivateKey(object):
+    @abc.abstractproperty
+    def key_size(self):
```

Comment:
This touches #1478 but I'm not sure what we can do about it here without also pulling that whole issue into scope for 0.8 (which I do not want to do).